go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -399,23 +545,33 @@ func (n *NodeController) ensureHybridOverlayBridge() error {
 	}
 
 	portName := houtil.GetHybridOverlayPortName(n.nodeName)
-
-	// If the master hasn't yet created our hybrid overlay port don't
-	// return an error, but allow the caller to try again later
-	portMAC, portIP, _ := util.GetPortAddresses(portName)
-	if portMAC == nil || portIP == nil {
+	portMACString, haveDRMACAnnotation := node.Annotations[hotypes.HybridOverlayDRMAC]
+	if !haveDRMACAnnotation {
+		klog.Warningf("node %s does not have DRMAC annotation yet, failed to ensure hybrid overlay"+
+			"and will retry later", n.nodeName)
+		// node must not be annotated yet, retry later
 		return nil
 	}
 
-	mgmtPortMAC, err := util.GetOVSPortMACAddress(util.K8sMgmtIntfName)
+	portMAC, err := net.ParseMAC(portMACString)
 	if err != nil {
-		return fmt.Errorf("failed to read management port MAC address: %v", mgmtPortMAC)
+		return fmt.Errorf("failed to parse DRMAC: %s", portMACString)
+	}
+	n.drMAC = portMAC
+
+	// n.drIP is always 3rd address in the subnet
+	// TODO add support for ipv6 later
+	portIP := subnet.IP.To4()
```

Comment:
Why is this not using `util.GetNodeHybridOverlayIfAddr()`? If there is a reason why that can't be used there needs to be a comment

Reply:
I did'nt know about that function, but now that I see it...it isn't very good. It calls get 2nd ip, which calls get first ip, which essentially just becomes IP Next.Next.Next. We should clean that up in a follow up patch and do it the way I'm doing it here.